lib/http-bridge/test/unit/Cardano/Wallet/HttpBridge/CompatibilitySpec.hs

```diff
@@ -123,6 +173,82 @@ spec = do
                     \8e3e95c150364fd775d4bc663ae6a9e6"
             hash `shouldBe` hash'
 
+    describe "Random Address Discovery Properties" $ do
+        it "isOurs works as expected during key derivation in testnet" $ do
+            property (prop_derivedKeysAreOurs @'Testnet)
+        it "isOurs works as expected during key derivation in mainnet" $ do
+            property (prop_derivedKeysAreOurs @'Mainnet)
+
+
+{-------------------------------------------------------------------------------
+                               Properties
+-------------------------------------------------------------------------------}
+prop_derivedKeysAreOurs
+    :: forall (n :: Network).
+       (KnownNetwork n, KeyToAddress (HttpBridge n) RndKey)
+    => Passphrase "seed"
+    -> Passphrase "encryption"
+    -> Index 'Hardened 'AccountK
+    -> Index 'Hardened 'AddressK
+    -> RndKey 'RootK XPrv
+    -> Property
+prop_derivedKeysAreOurs seed encPwd accIx addrIx rk' =
```

Comment:
Why is this in the `http-bridge` package :thinking: ? The `IsOurs` should be tested in the `core` package.

Reply:
I thought it is because of using `Mainnet and `Testnet which are defined : 
```
20  import Cardano.Wallet.HttpBridge.Compatibility                                                                                                                                            
21      ( HttpBridge, Network (..) )
```

Reply:
moreover, keyToAddress are implemented in `lib/http-bridge/src/Cardano/Wallet/HttpBridge/Compatibility.hs` and I used this function

Reply:
You can use: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/shared/Cardano/Wallet/DummyTarget/Primitive/Types.hs

If the test does not require an `http-bridge` specific functionality, there's no reason to have there :) 
Here, we are testing the random derivation scheme, so we could use a dummy target for this!

Reply:
I added this property to the core `AddressDiscovery.RandomSpec`.

There is a little more complexity because `instance KeyToAddress DummyTarget RndKey` still needs to do the derivation path encoding.